include/ghex/transport_layer/context.hpp

```diff
@@ -139,39 +31,35 @@ namespace gridtools {
                 using tag                    = TransportTag;
                 using transport_context_type = transport_context<tag,ThreadPrimitives>;
                 using communicator_type      = typename transport_context_type::communicator_type;
-                using parallel_context_type  = parallel_context<ThreadPrimitives>;
-                using thread_primitives_type = typename parallel_context_type::thread_primitives_type;
-                using thread_token           = typename parallel_context_type::thread_token;
+                using thread_primitives_type = ThreadPrimitives;
+                using thread_token           = typename thread_primitives_type::token;
 
             private:
-                parallel_context_type  m_parallel_context;
+                MPI_Comm m_mpi_comm;
+                thread_primitives_type m_thread_primitives;
                 transport_context_type m_transport_context;
 
             public:
                 template<typename...Args>
                 context(int num_threads, MPI_Comm comm, Args&&... args)
-                : m_parallel_context{num_threads, comm, std::forward<Args>(args)...}
-                , m_transport_context{m_parallel_context, comm, std::forward<Args>(args)...}
+                    : m_mpi_comm{comm}
+                    , m_thread_primitives(num_threads)
+                    , m_transport_context{m_thread_primitives, std::forward<Args>(args)...}
                 {}
 
                 context(const context&) = delete;
                 context(context&&) = delete;
 
             public:
 
-                const mpi_world& world() const noexcept 
+                /*const */thread_primitives_type& thread_primitives() /*const*/ noexcept
                 {
-                    return m_parallel_context.world(); 
-                }
-                
-                /*const */thread_primitives_type& thread_primitives() /*const*/ noexcept 
-                {
-                    return m_parallel_context.thread_primitives(); 
+                    return m_thread_primitives;
                 }
 
                 mpi::setup_communicator get_setup_communicator()
                 {
-                    return mpi::setup_communicator(m_parallel_context.world());
+                    return mpi::setup_communicator(m_mpi_comm);
```

Comment:
@boeschf Have a look at this one. Is this doing what it's supposed to now?

Reply:
this looks good now